src/cheri/riscv-priv-integration.adoc

```diff
@@ -767,105 +627,43 @@ Therefore, the behavior in this section isn't relevant if:
 
-In order to satisfy the definitions of such CSRs and preserve capability system invariants, the following procedure must be used as part of write-back to the CSR:
+To ensure that the bounds of a valid capability cannot be corrupted:
 
-. If A's address is invalid and A does not have infinite bounds (see
-xref:section_cap_encoding[xrefstyle=short]), then A's valid tag is set to zero.
-. Write the final (potentially modified) version of capability A to the CSR e.g.
-<<mtvecc>>, <<mepcc>>, etc.
+* If the capability's address is invalid and the capability does not have <<infinite-cap>> bounds, then set the valid tag to zero before writing to the CSR.
 
-NOTE: When A's address is invalid and happens to match an invalid address which the CSR
-can hold, then it is implementation defined whether to clear A's valid tag.
+NOTE: When the capability's address is invalid and happens to match an invalid address which the CSR
+can hold, then it is implementation defined whether to set the valid tag to zero.
 
 ==== Branches and Jumps
 
-Control transfer instructions jump or branch to a capability A which can be:
-
-* <<pcc>> for branches, direct jumps and any branch when in
-{cheri_int_mode_name} (see xref:section_cheri_hybrid_ext[xrefstyle=short]).
-* The capability in the *c* input register of a jump when in
-{cheri_cap_mode_name} (see xref:section_cheri_hybrid_ext[xrefstyle=short]).
-
-The following procedure must be used when jumping or branching to the target
-capability A if the <<pcc>> cannot hold all invalid addresses:
-
-. Calculate the effective target address T of the jump or branch as required by
-the instruction's behavior.
-. If T is invalid and A does not have infinite bounds (see
-xref:section_cap_encoding[xrefstyle=short]), then set A's valid tag to zero.
- This will cause a {cheri_excep_name_tag} when executing the target instruction.
-. If T is invalid and A has infinite bounds (see
-xref:section_cap_encoding[xrefstyle=short]), then A's valid tag is unchanged and T is
-written into A's address field. Attempting to execute the instruction at
-address T causes an instruction access fault or page fault as is usual in RISC-V.
-. Otherwise T is valid and the instruction behaves as normal.
-
-NOTE: RISC-V harts that do not support {cheri_base_ext_name} normally raise an
-instruction access fault or page fault after jumping or branching to an invalid address.
-Therefore, {cheri_base_ext_name} aims to preserve that behavior to ensure that
-harts supporting {cheri_base_ext_name} and {cheri_default_ext_name} are fully
-compatible with RISC-V harts provided that <<pcc>> and <<ddc>> are set to the
-<<infinite-cap>> capability.
+If the effective target address of the jump or branch is invalid, and the authorizing capability does not have <<infinite-cap>> bounds, then set the valid tag of the target <<pcc>> to zero.
+This will cause a {cheri_excep_name_pc} exception when executing the target instruction.
```

Comment:
```suggestion
This will cause a {cheri_excep_name_pc} exception when executing the target instruction.

NOTE: RISC-V harts that do not support {cheri_base_ext_name} normally raise an
instruction access fault or page fault after jumping or branching to an invalid address.
Therefore, {cheri_base_ext_name} aims to preserve that behavior to ensure that
harts supporting {cheri_base_ext_name} and {cheri_default_ext_name} are fully
compatible with RISC-V harts provided that <<pcc>> and <<ddc>> are set to the
<<infinite-cap>> capability.
```
Should we keep this note?